radixdlt/src/main/java/com/radixdlt/consensus/bft/VerifiedVertex.java

```diff
@@ -82,6 +86,18 @@ public Hash getParentId() {
 		return vertex.getQC().getProposed().getVertexId();
 	}
 
+
+	public interface PreparedVertexBuilder {
+		PreparedVertex andCommands(
+			ImmutableList<Command> successfulCommands,
+			ImmutableMap<Command, Exception> commandExceptions
+		);
+	}
+
+	public PreparedVertexBuilder withHeader(LedgerHeader ledgerHeader) {
```

Comment:
I appreciate that this is kind of cute, but not completely sold on the merits of this over a method called maybe `toPreparedVertex(header, successfulCommands, commandExceptions)`.

Reply:
It's really meant to separate two types of data:
1. LedgerHeader which gets voted on and eventually persisted
2. Other computed state which is transient

Agree not the best interface but should be good enough for now.